cEpiabm/src/python_bindings/bind_dataclasses.hpp

```diff
@@ -0,0 +1,85 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "dataclasses/infection_status.hpp"
+#include "dataclasses/compartment_counter.hpp"
+#include "population_factory.hpp"
+#include "toy_population_factory.hpp"
+#include "household_linker.hpp"
+
+namespace py = pybind11;
+
+void bind_dataclasses(py::module &m)
+{
+    using namespace epiabm;
+
+    py::enum_<InfectionStatus>(m, "InfectionStatus")
+        .value("Susceptible", InfectionStatus::Susceptible)
+        .value("Exposed", InfectionStatus::Exposed)
+        .value("InfectASympt", InfectionStatus::InfectASympt)
+        .value("InfectMild", InfectionStatus::InfectMild)
+        .value("InfectGP", InfectionStatus::InfectGP)
+        .value("InfectHosp", InfectionStatus::InfectHosp)
+        .value("InfectICU", InfectionStatus::InfectICU)
+        .value("InfectICURecov", InfectionStatus::InfectICURecov)
+        .value("Recovered", InfectionStatus::Recovered)
+        .value("Dead", InfectionStatus::Dead)
+        .export_values();
+
+
+    py::class_<PersonParams>(m, "PersonParams")
+        .def_readwrite("age", &PersonParams::age)
+        .def_readwrite("susceptibility", &PersonParams::susceptibility)
+        .def_readwrite("infectiousness", &PersonParams::infectiousness)
+        .def_readonly("next_status_time", &PersonParams::next_status_time);
+    py::class_<Person, PersonPtr>(m, "Person")
+        .def("status", &Person::status, py::return_value_policy::copy)
+        .def("params", &Person::params, py::return_value_policy::reference)
+        .def("cell_pos", &Person::cellPos, py::return_value_policy::copy)
+        .def("microcell_pos", &Person::microcellPos, py::return_value_policy::copy)
+        .def("microcell", &Person::microcell, py::return_value_policy::copy)
+        .def("places", &Person::places, py::return_value_policy::reference)
```

Comment:
Again a learning thing, but what is the distinction between the different `return_value_policy` values here?

Reply:
Normally return_value_policy is about ownership i.e. whether python or c++ will delete the object when there are no references to it.
Copy creates a new object in Python with the same value as in C++, and Python has full ownership.
Reference means Python can look at and modify the C++ variable, but when Python is finished with it it won't try to delete it (since it's still a C++ owned object)

There is also return_value_policy::take_ownership
Here the C++ object is given to Python, but Python takes responsibility for deleting it when it's finished with it.